atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -258,9 +333,9 @@ private static SerializableTransactionManager create(
                 transactionManager,
                 kvs,
                 sweepRunner,
-                Suppliers.ofInstance(config.enableSweep()),
-                Suppliers.ofInstance(config.getSweepPauseMillis()),
-                Suppliers.ofInstance(getSweepBatchConfig(config)),
+                () -> MoreObjects.firstNonNull(runtimeConfig.get().enableSweep(), config.enableSweep()),
+                () -> MoreObjects.firstNonNull(runtimeConfig.get().getSweepPauseMillis(), config.getSweepPauseMillis()),
+                () -> getSweepBatchConfig(runtimeConfig.get(), config),
```

Comment:
While I understand that the runtime parameters may change in the same run, I agree that this is not a big problem.
If we were to create a config, we would probably pass it in as a parameter of `new BackgroundSweeperImpl()`. But modifying this call would be a breaking change, since there a other products who call it directly.
We could still create a config and 'snapshot' the parameters before each run, but this also seems like a stretch for its value.